client/network/src/protocol.rs

```diff
@@ -349,6 +335,22 @@ impl<B: BlockT> BlockAnnouncesHandshake<B> {
 	}
 }
 
+/// Builds a SCALE-encoded "Status" message to send as handshake for the legacy protocol.
+fn build_status_message<B: BlockT>(protocol_config: &ProtocolConfig, chain: &Arc<dyn Client<B>>) -> Vec<u8> {
```

Comment:
Do I understand correctly that this function is called for each block that is imported even though we probably shake less than one hand per ~6 seconds? 

This decouples `protocol/generic_proto/behaviour.rs` from any upper layer logic, thus in case performance is irrelevant I prefer this push-approach over a pull-approach where the lower layer requests a new handshake message.

Reply:
> even though we probably shake less than one hand per ~6 seconds?

We do around two handshakes per second at the moment.

Reply:
> I prefer this push-approach over a pull-approach where the lower layer requests a new handshake message

The problem with this "push approach" which consists in holding a copy of the handshake message on each background task is that we have to send one message to each background task, which causes all the background tasks to wake up, and clogs a bit more the channel between the worker and tasks.